spec/requests/doctors_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
+RSpec.describe 'doctor Api', type: :request do
+  before :each do
+    post '/users', params: {
+      user: {
+        username: 'John Doe',
+        email: 'test@email.com',
+        password: 'Password1'
+      }
+    }.to_json, headers: {
+      'Content-Type': 'application/json',
+      Accept: 'application/json'
+    }
+  end
+
```

Comment:
Nice implementing the the user to log in

Reply:
Thank you, for four feedback.